jaxley/modules/base.py

```diff
@@ -2792,6 +2864,37 @@ def _set_inds_in_view(
                 self._edges_in_view = np.intersect1d(
                     possible_edges_in_view, self._edges_in_view
                 )
+            base_comp_edges = self.base._comp_edges
+            if not base_comp_edges.empty:
+                pre = base_comp_edges["source"].isin(incl_comps).to_numpy()
+                post = base_comp_edges["sink"].isin(incl_comps).to_numpy()
+                comp_edge_inds = base_comp_edges.index.to_numpy()
+                if comp_edge_condition == "source_and_sink":
+                    possible_edges_in_view = comp_edge_inds[(pre & post).flatten()]
+                elif comp_edge_condition == "source_or_sink":
+                    possible_edges_in_view = comp_edge_inds[(pre | post).flatten()]
+                elif comp_edge_condition == "startpoint":
+                    # Type 2 and 4 are comp_edges between branchpoints and the
+                    # child compartments.
+                    is_child = base_comp_edges["type"].isin([2, 4]).to_numpy()
+                    possible_edges_in_view = comp_edge_inds[
+                        ((pre | post) & is_child).flatten()
+                    ]
+                elif comp_edge_condition == "endpoint":
+                    # Type 2 and 4 are comp_edges between branchpoints and the
+                    # parent compartments.
+                    is_parent = base_comp_edges["type"].isin([1, 3]).to_numpy()
+                    possible_edges_in_view = comp_edge_inds[
+                        ((pre | post) & is_parent).flatten()
+                    ]
+                else:
+                    raise ValueError(
+                        f"comp_edge_condition is {comp_edge_condition}, but must be in "
+                        "{source_and_sink, source_or_sink, startpoint, endpoint}."
+                    )
+                self._comp_edges_in_view = np.intersect1d(
+                    possible_edges_in_view, self._comp_edges_in_view
+                )
```

Comment:
it might make sense to seperate this out into an extra function ala `_set_comp_edges_in_view` for clearity

Reply:
...and add a doc string